packages/nvidia_nat_a2a/tests/server/test_front_end_config_security.py

```
@@ -0,0 +1,78 @@
+        assert "without authentication" in str(exc_info.value)
+
+    def test_non_localhost_bind_rejects_named_host(self):
+        """A named host like 'agent.example.com' without server_auth also fails."""
+        with pytest.raises(ValidationError) as exc_info:
+            A2AFrontEndConfig(host="agent.example.com")
+        assert "without authentication" in str(exc_info.value)
+
+    def test_non_localhost_bind_allowed_with_explicit_ack(self):
+        """Operators who run behind an external auth layer can opt in explicitly."""
+        cfg = A2AFrontEndConfig(
+            host="0.0.0.0",
```

Ruff (0.15.10)

[error] Possible binding to all interfaces

(S104)